### PR TITLE
feat: [#188443179] Unsuccessful Business Address & Zipcode License Check Immediate Error State

### PR DESCRIPTION
**Pull request by cbrunefearless**
## Description
User Story
As a user that has conducted a license status check search, I want to be informed that no license has been found immediately when viewing the task and the error message persists until the information is updated.

Acceptance Criteria
Given a license search have been conducted
and the check did not find a license
when viewing the license task
then the 'License could not be found..." warning is rendered in the task
and the error persists

dev note: the license status of UNKNOWN is used to trigger the alert, the status changes should be handled in the backend

### Ticket

This pull request resolves [#188443179](https://www.pivotaltracker.com/story/show/188443179).

## Code author checklist

- [X] I have rebased this branch from the latest main branch
- [X] I have performed a self-review of my code
- [X] I have created and/or updated relevant documentation on the engineering documentation website
- [X] I have not used any relative imports
- [X] I have pruned any instances of unused code
- [X] I have not added any markdown to labels, titles and button text in config
- [X] If I added/updated any values in `userData` (including `profileData`, `formationData` etc), then I added a new migration file
- [X] I have checked for and removed instances of unused config from CMS
- [X] If I added any new collections to the CMS config, then I updated the search tool and `cmsCollections.ts` (see CMS Additions in Engineering Reference/FAQ on the engineering documentation site)
- [X] I have updated relevant `.env` values in both `.env-template` and in Bitwarden

### CODE REVIEW

File: api/src/api/licenseStatusRouter.test.ts
Comment (FarazA22) [inline api/src/api/licenseStatusRouter.test.ts:61]: sand / pebble - comment is no longer needed

File: api/src/client/WebserviceLicenseStatusProcessorClient.test.ts
Comment (FarazA22) [inline api/src/client/WebserviceLicenseStatusProcessorClient.test.ts:481]: boulder - you should review how other error are throw and update test accordingly, there are other examples under webflow and dynamics. There should be test not only at the client level, but also each level the parent function is called to ensure the error is traveling back/ user gets a response.

File: api/src/api/licenseStatusRouter.test.ts
Comment (cbrunefearless) [reply api/src/api/licenseStatusRouter.test.ts:61]: done

File: api/src/client/WebserviceLicenseStatusProcessorClient.ts
Comment (FarazA22) [inline api/src/client/WebserviceLicenseStatusProcessorClient.ts:102]: Pebble - how are we logging this?

File: api/src/client/WebserviceLicenseStatusProcessorClient.test.ts
Comment (cbrunefearless) [reply api/src/client/WebserviceLicenseStatusProcessorClient.test.ts:264]: done

File: shared/src/license.ts
Comment (FarazA22) [inline shared/src/license.ts:72]: boulder - Do we need a migration for this? If not, can you explain why not.

File: web/src/components/tasks/LicenseTask.test.tsx
Comment (FarazA22) [inline web/src/components/tasks/LicenseTask.test.tsx:109]: pebble - we should keep license data, I'd say this, because there is logic tied to lastUpdatedISO within LicenseData, we provide lastUpdatedISO within each license result.

File: api/src/domain/user/updateLicenseStatusFactory.test.ts
Comment (cbrunefearless) [reply api/src/domain/user/updateLicenseStatusFactory.test.ts:274]: renamed the tests that have changed.

File: web/src/components/tasks/LicenseTask.test.tsx
Comment (FarazA22) [inline web/src/components/tasks/LicenseTask.test.tsx:127]: boulder - there are multiple instances of lastUpdatedISO, due to this change, the test is passing for the wrong reason.

File: web/src/components/tasks/LicenseTask.test.tsx
Comment (FarazA22) [inline web/src/components/tasks/LicenseTask.test.tsx:448]: Pebble - it would make more sense to provide license data for tasks that are not relevant to the page and confirm the alert appears. The idea is other data can be retrieved and the alert should appear for tasks that are not relevant. It you keep the existing changes then further tests are needed.

File: web/src/components/tasks/LicenseTask.tsx
Comment (FarazA22) [inline web/src/components/tasks/LicenseTask.tsx:57]: pebble - `hasDoneLicenseCheck` doesn't need to exist. you should consider the logic based on the 2by2 matrix exercise we did yesterday to figure out all the possible outcomes.

File: web/src/components/tasks/LicenseTask.tsx
Comment (FarazA22) [inline web/src/components/tasks/LicenseTask.tsx:62]: pebble - I think it makes more sense for this to be a else statement

File: web/src/components/tasks/LicenseTask.tsx
Comment (FarazA22) [inline web/src/components/tasks/LicenseTask.tsx:96]: sand - you need to figure out if this function needs to be delete or placed in another area.

File: api/src/client/WebserviceLicenseStatusProcessorClient.ts
Comment (cbrunefearless) [reply api/src/client/WebserviceLicenseStatusProcessorClient.ts:102]: Added logging

File: web/src/components/tasks/LicenseTask.tsx
Comment (FarazA22) [inline web/src/components/tasks/LicenseTask.tsx:96]: typo

Comment (cbrunefearless) [reply web/src/components/tasks/LicenseTask.tsx:96]: taken out

File: web/src/components/tasks/LicenseTask.tsx
Comment (FarazA22) [inline web/src/components/tasks/LicenseTask.tsx:55]: take a look at the initial of license data, it's initially set to undefined, then when the search is done, it's replaced with a object

File: web/src/components/tasks/LicenseTask.tsx
Comment (FarazA22) [inline web/src/components/tasks/LicenseTask.tsx:97]: typo

File: web/src/components/tasks/LicenseTask.tsx
Comment (FarazA22) [inline web/src/components/tasks/LicenseTask.tsx:96]: typo

File: web/src/lib/api-client/apiClient.test.ts
Comment (FarazA22) [inline web/src/lib/api-client/apiClient.test.ts:67]: pebble - you should review the test and determine if they should be renamed, rewritten, or deleted.

File: web/src/lib/api-client/apiClient.test.ts
Comment (FarazA22) [inline web/src/lib/api-client/apiClient.test.ts:71]: pebble - you should review the test and determine if they should be renamed, rewritten, or deleted.

File: web/src/components/tasks/LicenseTask.test.tsx
Comment (FarazA22) [reply web/src/components/tasks/LicenseTask.test.tsx:109]: I just realized your using lastUpdatedISO that is defined withing the business object

File: api/src/api/licenseStatusRouter.test.ts
Comment (somabadri) [inline api/src/api/licenseStatusRouter.test.ts:55]: [pebble/dust] pardon my confusion, where in this test are we asserting that licenseData is an empty object?

File: api/src/domain/user/updateLicenseStatusFactory.ts
Comment (somabadri) [inline api/src/domain/user/updateLicenseStatusFactory.ts:123]: [pebble] make sure to delete these before merging

File: web/src/components/tasks/LicenseTask.tsx
Comment (somabadri) [inline web/src/components/tasks/LicenseTask.tsx:97]: [pebble] console log statements which need to be removed

File: web/src/components/tasks/LicenseTask.test.tsx
Comment (somabadri) [inline web/src/components/tasks/LicenseTask.test.tsx:462]: [pebble] How does the `checkLicenseStatus` function return an empty string? The empty object is returned on line 96 in `updateLicenseStatusFactory` and that uses `updateLicenseStatusAndLicenseTask` function which sets `lastUpdatedISO`to currentDate on line 58? Let me know if I'm getting this confused.

File: api/src/client/WebserviceLicenseStatusProcessorClient.test.ts
Comment (jphechter) [inline api/src/client/WebserviceLicenseStatusProcessorClient.test.ts:311]: [sand]
```suggestion
  it("throws an error when status is not in list of known License Statuses", async () => {
```

File: api/src/api/licenseStatusRouter.test.ts
Comment (FarazA22) [inline api/src/api/licenseStatusRouter.test.ts:55]: sand - i'd remove the text `from a license task` since this update can occur from having the address in profile

File: api/src/client/WebserviceLicenseStatusProcessorClient.test.ts
Comment (FarazA22) [inline api/src/client/WebserviceLicenseStatusProcessorClient.test.ts:26]: sand - can the variable name be updated to dummyLogger or mockLogger so it's clear what is being used

File: api/src/client/WebserviceLicenseStatusProcessorClient.test.ts
Comment (FarazA22) [inline api/src/client/WebserviceLicenseStatusProcessorClient.test.ts:23]: Boulder - Since `updateLicenseStatusFactory`, `userRouterFactory`, and `licenseStatusRouterFactory` use this function, I'd expect each to have a test that validates the error from the line `throw new Error(Cannot find license status ${value})` is thrown.

File: api/src/client/WebserviceLicenseStatusProcessorClient.test.ts
Comment (FarazA22) [inline api/src/client/WebserviceLicenseStatusProcessorClient.test.ts:308]: r

File: api/src/client/WebserviceLicenseStatusProcessorClient.test.ts
Comment (FarazA22) [inline api/src/client/WebserviceLicenseStatusProcessorClient.test.ts:318]: dust - for consistency sake, this should be Fakestatus?

File: api/src/client/WebserviceLicenseStatusProcessorClient.ts
Comment (FarazA22) [inline api/src/client/WebserviceLicenseStatusProcessorClient.ts:104]: Boulder - This message sent to the logger is no consistent, how would I be able to debug this? What information is missing?

File: api/src/db/migrations/v144_add_raffle_bingo_games.ts
Comment (FarazA22) [inline api/src/db/migrations/v144_add_raffle_bingo_games.ts:811]: Pebble - all the generators in this file generate a random string, what's the though on using empty strings? license statues should also be a random value / or randomly selected from the list of possible statuses

File: api/src/db/migrations/v145_remove_haserror_field_from_licensedetails.test.ts
Comment (FarazA22) [inline api/src/db/migrations/v145_remove_haserror_field_from_licensedetails.test.ts:10]: solid test, nice job

File: api/src/db/migrations/v145_remove_haserror_field_from_licensedetails.ts
Comment (FarazA22) [inline api/src/db/migrations/v145_remove_haserror_field_from_licensedetails.ts:29]: pebble - I think line 30 should come first?

File: api/src/db/migrations/v145_remove_haserror_field_from_licensedetails.ts
Comment (FarazA22) [inline api/src/db/migrations/v145_remove_haserror_field_from_licensedetails.ts:37]: pebble - this should return `Partial<Record<v144LicenseName, v144LicenseDetails>>`

File: api/src/db/migrations/v145_remove_haserror_field_from_licensedetails.ts
Comment (FarazA22) [inline api/src/db/migrations/v145_remove_haserror_field_from_licensedetails.ts:38]: pebble - v144Licenses exist in the other file and can be imported, the v145Licenses needs to be created

Comment (FarazA22) [inline api/src/db/migrations/v145_remove_haserror_field_from_licensedetails.ts:38]: pebble - v144Licenses exist in the other file and can be imported, you should reference v145Licenses since it exist... Not Partial<Record<v144LicenseName, v144LicenseDetails>>

File: api/src/db/migrations/v145_remove_haserror_field_from_licensedetails.ts
Comment (FarazA22) [inline api/src/db/migrations/v145_remove_haserror_field_from_licensedetails.ts:49]: Pebble - we should not use as, can you update this.

**Review by FarazA22 — CHANGES_REQUESTED** [review]
test

File: api/src/db/migrations/v145_remove_haserror_field_from_licensedetails.ts
Comment (FarazA22) [inline api/src/db/migrations/v145_remove_haserror_field_from_licensedetails.ts:44]: Pebble - we should not use as, can you update this.

File: api/src/domain/user/updateLicenseStatusFactory.test.ts
Comment (FarazA22) [inline api/src/domain/user/updateLicenseStatusFactory.test.ts:289]: sand - update to `returns empty license object`

File: web/src/components/tasks/LicenseTask.test.tsx
Comment (FarazA22) [inline web/src/components/tasks/LicenseTask.test.tsx:452]: dust - based on you b/e logic, we'd return an empty object , or licenses that are not tied to the current task, therefore ,shouldn't this be an empty object?

File: web/src/components/tasks/LicenseTask.tsx
Comment (FarazA22) [inline web/src/components/tasks/LicenseTask.tsx:57]: Nice

File: web/src/components/tasks/LicenseTask.test.tsx
Comment (FarazA22) [inline web/src/components/tasks/LicenseTask.test.tsx:448]: We should have a test at the take page level where you have license data for license A, but visiting the task page for license B, the check had been done so you see the hot found error

**Pull request merged**
